File: kbs_bbs/bbs2www/html/bbsfoot.php
<?php
    $setboard=0;
    require("funcs.php");
    if ($loginok != 1)
        html_nologin();
    else {
        html_init("gb2312","","<meta http-equiv=\"Refresh\" content=\"240; url=/bbsfoot.php\">",1);

		if (isset($_GET["total"]))
			$oldtotal = $_GET["total"];
		else
			$oldtotal = 0;
		settype($oldtotal,"integer");

		if (isset($_GET["unread"]))
			$oldunread = $_GET["unread"];
		else
			$oldunread = 0;
		settype($oldunread,"integer");
?>
<script language="JavaScript">
<!--
function Init() {
  servertime=new Date()
  servertime.setTime(<?php echo time(); ?>*1000)
  staytime=<?php echo (time()-$currentuinfo["logintime"])/60; ?>

  localtime=new Date()
  Time()
}
function Time(){
 var now=new Date()
 var Timer=new Date()
 Timer.setTime(servertime.getTime()+now.getTime()-localtime.getTime());
 var hours=Timer.getHours()
 var minutes=Timer.getMinutes()
 if (hours==0)
 hours=12
 if (minutes<=9)
 minutes="0"+minutes
 var year=Timer.getYear();
 if (year < 1900)   
	 year = year + 1900; 
 myclock=year+"年"+(Timer.getMonth()+1)+"月"+Timer.getDate()+"日"+hours+":"+minutes
 var staysec=(now.getTime()-localtime.getTime())/60000+staytime;
 stayclock=parseInt(staysec/60)+"小时"+parseInt(staysec%60)+"分钟"
 document.clock.myclock.value=myclock
 document.clock.stay.value=stayclock
 setTimeout("Time()",58000)
}
//JavaScript End-->
</script>
<body class="b2"  onload="Init()">
<form name="clock">时间[<input class="b8" TYPE="text" NAME="myclock" size="18">] 在线[<a href="/cgi-bin/bbs/bbsusr" target="f3" class="b8"><?php 
echo bbs_getonlinenumber(); ?></a>] 帐号[<a href=<?php
echo "\"/bbsqry.php?userid=" . $currentuser["userid"] . "\""; ?> target="f3" class="b8"><?php
echo $currentuser["userid"]; ?></a>] <?php
		if (strcmp($currentuser["userid"], "guest") != 0)
		{
		    if (bbs_getmailnum($currentuser["userid"],$total,$unread, $oldtotal, $oldunread)) {
			  if ($unread!=0) {
		        echo "信箱[<a href=\"bbsreadmail.php\" target=\"f3\" class=\"b8\">" . $total . "封(新信" . $unread . ")</a>] ";
			  }
			  else {
		        echo "信箱[<a href=\"bbsreadmail.php\" target=\"f3\" class=\"b8\">" . $total . "封</a>] ";
			  }
			}
			else
		      echo "信箱[<a href=\"bbsreadmail.php\" target=\"f3\" class=\"b8\">0封</a>] ";
		}
?>
停留[<input class="b8" TYPE="text" NAME="stay" size="10">]
</form>
<?php
        html_normal_quit();
    }
?>

File: kbs_bbs/bbs2www/src/bbsfoot.c
/*
 * $Id$
 */
#include "bbslib.h"

int main()
{
	int dt=0, mail_total, mail_unread;
	char *id="guest";

	init_all();
	if(loginok)
	{
		id=currentuser->userid;
		dt=abs(time(0) - u_info->logintime)/60;
				set_idle_time(u_info, time(0));
	}
	printf("<script language=\"JavaScript\">\n"
"<!--\n"
"function Init() {\n"
"  servertime=new Date()\n"
"  servertime.setTime(%d*1000)\n"
"  staytime=%d\n"
"  localtime=new Date()\n"
"  Time()\n"
"}\n"
"function Time(){\n"
" var now=new Date()\n"
" var Timer=new Date()\n"
" Timer.setTime(servertime.getTime()+now.getTime()-localtime.getTime());\n"
" var hours=Timer.getHours()\n"
" var minutes=Timer.getMinutes()\n"
" if (hours==0)\n"
" hours=12\n"
" if (minutes<=9)\n"
" minutes=\"0\"+minutes\n"
" var year=Timer.getYear();\n"
" if (year < 1900)   \n"
"	 year = year + 1900; \n"

" myclock=year+\"年\"+(Timer.getMonth()+1)+\"月\"+Timer.getDate()+\"日\"+hours+\":\"+minutes\n"
" var staysec=(now.getTime()-localtime.getTime())/60000+staytime;\n"
" stayclock=parseInt(staysec/60)+\"小时\"+parseInt(staysec%60)+\"分钟\"\n"
" document.clock.myclock.value=myclock\n"
" document.clock.stay.value=stayclock\n"
" setTimeout(\"Time()\",58000)\n"
"}\n"
"//JavaScript End-->\n"
"</script>",time(0)+20,loginok?dt:0);
 /*
if (document.layers){\n\
document.layers.position.document.write(myclock)\n\
document.layers.position.document.close()\n\
document.layers.stay.document.write(stayclock)\n\
document.layers.stay.document.close()\n\
}\n\
else if (document.all)\n\
position.innerHTML=myclock\n\
stay.innerHTML=stayclock\n\
setTimeout(\"Time()\",58000)\n\
 }\n\
 */

	printf("<style type=\"text/css\">\nA {color: #0000FF}\n</style>\n");
  	printf("<body onload=\"Init()\"><form name=\"clock\">");
  	printf("时间[<INPUT class=\"readonly\" TYPE=\"text\" NAME=\"myclock\" size=\"18\">] ");
	/*
  	printf("时间[<span id=\"myclock\"></span>] ");
	*/
	printf("在线[<a href=\"bbsusr\" target=\"f3\">%d</a>] ", count_online());
	printf("帐号[<a href=\"bbsqry?userid=%s\" target=\"f3\">%s</a>] ", id, id);
	if(loginok)
	{
		mail_total=mails(id, 0);
		mail_unread=mails(id, 1);
		if(mail_unread==0)
		{
			printf("信箱[<a href=\"bbsmail\" target=\"f3\">%d封</a>] ",
					mail_total);
		}
		else
		{
			printf("信箱[<a href=\"bbsmail\" target=\"f3\">%d(新信<font color=\"red\">%d</font>)</a>] ", 
				mail_total, mail_unread);
		}
	}
	printf("停留[<INPUT class=\"readonly\" TYPE=\"text\" NAME=\"stay\" size=\"10\">]</form>");
	/*
	printf("停留[<span id=\"stay\"></span>]");
	*/
  	printf("</body></html>");
}

int mails(char *id, int unread_only) {
        struct fileheader x;
        char path[80];
	int total=0, unread=0;
        FILE *fp;
	if(!loginok) return 0;
        sprintf(path, "mail/%c/%s/.DIR", toupper(id[0]), id);
        fp=fopen(path, "r");
        if(fp==0) return 0;
        while(fread(&x, sizeof(x), 1, fp)>0) {
                total++;
                if(!(x.accessed[0] & FILE_READ)) unread++;
        }
        fclose(fp);
	if(unread_only) return unread;
	return total;
}

